### DISCUSSION TITLE
What about rename axmol::Ref to axmol::Object

### DISCUSSION THREAD

**Discussion by halx99** [axmolengine/axmol#1858] [General]
RT

**Comment by rh101** [comment]
It would be a straightforward change for developers using Axmol, not breaking anything, so I don't see any reason against it.

**Comment by smilediver** [comment]
I don't have a strong preference, so I would just leave `Ref`. And it's not like C#, where every object would inherit it. Also, then `RefPtr` would sound weird. If `Ref` was `Object` and `RefPtr` was `Ref` from the start, then maybe, but changing now would bring a lot of chaos.

**Reply by halx99** [reply]
RefPtr should be instruive_ptr

**Reply by smilediver** [reply]
You mean `RefPtr` should be renamed to `intrusive_ptr`? If so, that would only increase the code bloat without any added value. I'd keep it short.

**Reply by halx99** [reply]
OK, let's keep RefPtr
